File: backend/app/models/assistant.py
from datetime import date
from enum import Enum
from typing import List, Optional, Union

from pydantic import BaseModel

class AssistantItem(BaseModel):
    id: Optional[int] = None
    name: str = ""
    quantity: int = 0
    notes: Optional[str] = None

class AssistantCategory(BaseModel):
    id: Optional[int] = None
    name: str = ""
    items: List[AssistantItem] = []

class AssistantTrip(BaseModel):
    id: Optional[int] = None
    name: Optional[str] = None
    description: Optional[str] = None
    start_date: Optional[date] = None
    end_date: Optional[date] = None

class StartAssistantQuery(BaseModel):
    trip_id: Optional[int] = None

class StartAssistantResponse(BaseModel):
    message: str
    trip: AssistantTrip
    categories: List[AssistantCategory]
    uncategorized_items: List[AssistantItem]

class ChatAssistantRequest(BaseModel):
    user_msg: str
    trip: AssistantTrip
    categories: List[AssistantCategory]
    uncategorized_items: List[AssistantItem]

class ChatAssistantResponseMode(str, Enum):
    MESSAGE = "message"
    VALUES = "values"

class ChatAssistantMessageRole(str, Enum):
    HUMAN = "human"
    AI = "ai"

class ChatAssistantMessage(BaseModel):
    type: ChatAssistantMessageRole
    content: str

class ChatAssistantValues(BaseModel):
    messages: List[ChatAssistantMessage]
    trip: AssistantTrip
    categories: List[AssistantCategory]
    uncategorized_items: List[AssistantItem]

class ChatAssistantResponse(BaseModel):
    done: bool = False
    mode: ChatAssistantResponseMode
    content: Union[str, ChatAssistantValues]

class AcceptAssistantRequest(BaseModel):
    trip: AssistantTrip
    categories: List[AssistantCategory]
    uncategorized_items: List[AssistantItem]

class AcceptAssistantResponse(BaseModel):
    trip_id: int

File: frontend/types/assistant.ts
export interface AssistantTrip {
    id?: number
    name?: string
    description?: string
    start_date?: string
    end_date?: string
}

export interface AssistantCategory {
    id?: number
    name: string
    items: AssistantItem[]
}

export interface AssistantItem {
    id?: number
    name: string
    notes?: string
    quantity: number
}

export interface AssistantMessage {
    role: "user" | "assistant"
    content: string
}

export interface StartAssistantRequest {
    tripId?: number
}

export interface StartAssistantResponse {
    message: string
    trip: AssistantTrip
    categories: AssistantCategory[]
    uncategorized_items: AssistantItem[]
}

export interface ChatAssistantRequest {
    user_msg: string
    trip: AssistantTrip
    categories: AssistantCategory[]
    uncategorized_items: AssistantItem[]
}

export enum ChatAssistantMode {
    MESSAGE = "message",
    VALUES = "values"
}

export interface ChatAssistantValues {
    trip: AssistantTrip
    categories: AssistantCategory[]
    uncategorized_items: AssistantItem[]
}

export interface ChatAssistantResponse {
    done: boolean
    mode: ChatAssistantMode
    content: string | ChatAssistantValues
}

export interface AcceptAssistantRequest {
    trip: AssistantTrip
    categories: AssistantCategory[]
    uncategorized_items: AssistantItem[]
}

export interface AcceptAssistantResponse {
    trip_id: number
}
